network/src/protocols/support_protocols.rs

```diff
@@ -0,0 +1,134 @@
+use crate::ProtocolId;
+use p2p::{
+    builder::MetaBuilder,
+    service::{BlockingFlag, ProtocolHandle, ProtocolMeta},
+    traits::ServiceProtocol,
+};
+use tokio_util::codec::length_delimited;
+
+pub enum SupportProtocols {
```

Comment:
FYI. `SupportProtocols` is just like a "constants wrapper", which wraps the protocols and their contributions. It seems that the protocols and contributions are pre-defined. I think the plain style would be more intuitive(define as constants):

  ```
  pub const PING_PROTOCOL_NAME = "ping";
  pub const PING_SUPPORTED_VERSIONS = vec!["1"];
  ```